api/types/DirectMessageThread.js

```diff
@@ -31,6 +31,7 @@ const DirectMessageThread = /* GraphQL */ `
 
 	extend type Query {
 		directMessageThread(id: ID!): DirectMessageThread
+		directMessageThreadByUserId(userId: ID!): DirectMessageThread
```

Comment:
Does this need a separate query? Imo querying with an object that has either an `id` field or an `userId` field makes a lot more sense, if that's possible. Maybe something like this works?

```GraphQL
extend type Query {
  type DirectMessageThreadQueryInput {
    id: ID
    userId: ID
  }
  directMessageThread(input: ID! | DirectMessageThreadQueryInput): DirectMessageThread
}
```

Reply:
While that *is* a possible way to accomplish this, I'm honestly not sure if it's better. I feel that a common mistake we've made in our queries and mutations is to try and be too clever by bundling multiple possible intentions inside a single action. For example we have `toggleChannelSubscription` instead of two: `subscribeToChannel` and `unsubscribeFromChannel` or something like that - as a result, our mutation is much much more complex.

In the same way, I worry that trying to do two things at once here may be confusing.

>  instead, what would have been ideal is for us to simply let the addDirectMessage mutation create the thread on the server if no thread id exists in the mutation variables.

^^ I said this above, but feel less good about that idea now with this in mind. It would require over-complicating the input variables that the mutation requires. I'd much prefer to have a strong set of single-purpose queries and mutations that are easy to read and refactor, versus trying to be clever and combine multiple use cases in one.

What do you think?

Reply:
Talked offline and agreed :)